astronomer/providers/dbt/cloud/operators/dbt.py

```diff
@@ -0,0 +1,80 @@
+import time
+from typing import Any, Dict, Optional
+
+from airflow import AirflowException
+from airflow.providers.dbt.cloud.hooks.dbt import DbtCloudHook
+from airflow.providers.dbt.cloud.operators.dbt import DbtCloudRunJobOperator
+from airflow.utils.context import Context
+
+from astronomer.providers.dbt.cloud.triggers.dbt import DbtCloudRunJobTrigger
+
+
+class DbtCloudRunJobOperatorAsync(DbtCloudRunJobOperator):
+    """
+    Executes a dbt Cloud job asynchronously. Trigger the dbt cloud job via worker to dbt and with run id in response
+    poll for the status in trigger.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:DbtCloudRunJobOperator`
+
+    :param dbt_cloud_conn_id: The connection ID for connecting to dbt Cloud.
+    :param job_id: The ID of a dbt Cloud job.
+    :param account_id: Optional. The ID of a dbt Cloud account.
+    :param trigger_reason: Optional. Description of the reason to trigger the job.
+    :param steps_override: Optional. List of dbt commands to execute when triggering the job instead of those
+        configured in dbt Cloud.
+    :param schema_override: Optional. Override the destination schema in the configured target for this job.
+    :param wait_for_termination: This Flag is disabled in this async operator.wait_for_termination
+        will be always set True
+    :param timeout: Time in seconds to wait for a job run to reach a terminal status for non-asynchronous
+        waits. Used only if ``wait_for_termination`` is True. Defaults to 7 days.
+    :param check_interval: Time in seconds to check on a job run's status for non-asynchronous waits.
+        Used only if ``wait_for_termination`` is True. Defaults to 60 seconds.
+    :param additional_run_config: Optional. Any additional parameters that should be included in the API
+        request when triggering the job.
+    :return: The ID of the triggered dbt Cloud job run.
+    """
+
+    def execute(self, context: "Context") -> None:  # type: ignore[override]
+        """Submits a job which generates a run_id and gets deferred"""
+        hook = DbtCloudHook(dbt_cloud_conn_id=self.dbt_cloud_conn_id)
+        trigger_job_response = hook.trigger_job_run(
+            account_id=self.account_id,
+            job_id=self.job_id,
+            cause=self.trigger_reason,
```

Comment:
cause is defined as a required param in the trigger_job_run method and it needs to be string, but trigger_reason seems to optional in the docstring.

Reply:
cause is required while making call it was giving me error, we need to fix it OSS as well

Reply:
It's optional because if it's not provided, the original `DbtCloudRunJobOperator` [sets a default reason](https://github.com/apache/airflow/blob/23ad7e292a29e43436170078ae132569f8fbf1e4/airflow/providers/dbt/cloud/operators/dbt.py#L117). That logic is carried over here though.

Perhaps the docstring in both here and in OSS could be clearer about that.

Reply:
Corrected the doc string and added code to check for the `trigger_reason` if its not there add the default reason